packages/nx-monorepo/src/nx-monorepo.ts

```diff
@@ -599,6 +601,60 @@ export class NxMonorepoProject extends TypeScriptProject {
     }
   }
 
+  /**
+   * Adds a dependency between two Java Projects in the monorepo.
+   * @param dependent project you want to have the dependency
+   * @param dependee project you wish to depend on
+   */
+  public addJavaDependency(dependent: JavaProject, dependee: JavaProject) {
+    // Add implicit dependency for build order
+    this.addImplicitDependency(dependent, dependee);
+
+    // Add dependency in pom.xml
+    dependent.addDependency(
+      `${dependee.pom.groupId}/${dependee.pom.artifactId}@${dependee.pom.version}`
+    );
+
+    // Add a repository so that the dependency in the pom can be resolved
+    dependent.pom.addRepository({
```

Comment:
This will be OK even though the dep may not have dist/java? I assume this only matters as part of build as java doesn't have an install task?

Reply:
Good callout! Yeah maven would fail at build time. From some more digging it turns out maven packaging is always on for JavaProjects, but you can customise this dist dir. Sadly it's not exposed at the moment so I've raised this PR: https://github.com/projen/projen/pull/2624

Happy to wait until that gets merged and update, or to just assume dist/java and revisit when the dist dir is accessible :)

Reply:
Let's wait and fix it all at once :)

Also - your linked PR is failing as I assume you didn't do a full rebuild of projen.

Reply:
As discussed offline, we'll assume `dist/java` for now but update when the projen PR is merged. Tracking here: https://github.com/aws/aws-prototyping-sdk/issues/362